src/scripts/exclusion_reasons_enum_updater.py

```diff
@@ -0,0 +1,95 @@
+"""Updates exclusion reasons enum: src/schema/mondo.yaml
+
+Resources:
+- Source CSV: https://docs.google.com/spreadsheets/d/18V-_j4b22LzCytxFEM3pBEl6-_ODJY06gpUwH0F6JVY/edit#gid=1644570180
+
+Instructions:
+1. Save the 'exclusion code and definitions' tab from the above GoogleSheet as `src/scripts/exclusion_reasons.csv`
+2. From `src/ontology/`, run: `sh run.sh make update-exclusion-reasons`
+"""
+import os
+import re
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict
+
+import pandas as pd
+import yaml
+
+
+GSHEETS_URL = 'https://docs.google.com/spreadsheets/d/18V-_j4b22LzCytxFEM3pBEl6-_ODJY06gpUwH0F6JVY/edit#gid=1644570180'
+PROJECT_DIR = Path(os.path.dirname(__file__)).parent.parent
+SHEET_YML_MAPPINGS = {
+	'code_name': 'title',
+	'definitions': 'description',
+	'notes': 'notes',
+	'examples': 'examples',
+	'exclude_children': 'exclude_children',
+}
+DEFAULTS = {
+	'input_path_exclusion_reasons': os.path.join(PROJECT_DIR, 'src', 'scripts', 'exclusion_reasons.csv'),
+	'input_path_mondo_schema': os.path.join(PROJECT_DIR, 'src', 'schema', 'mondo.yaml'),
+}
+
+
+def run(input_path_exclusion_reasons: str, input_path_mondo_schema: str):
+	"""Runs script"""
+	# Load updated exclusions
+	# - Drops all rows where all cells are empty
+	df = pd.read_csv(input_path_exclusion_reasons).dropna(axis=0, how='all').fillna('')
+	# - Format to remove whitespace, leading indentation markers, and unneeded MONDO prefix
+	df['codes'] = df['codes'].apply(lambda x: re.sub('.*MONDO:', '', x))
+	# - Strip all whitespace
+	df = df.apply(lambda x: x.str.strip() if x.dtype == 'object' else x)
+
+	# Load current exclusions
+	with open(input_path_mondo_schema, 'r') as stream:
+		mondo_schema = yaml.safe_load(stream)
+	exclusion_dicts: Dict[str, Dict[str, str]] = mondo_schema['enums']['entity_type_enum']['permissible_values']
+
+	# Delete: Any schema enum items not located in GoogleSheet source of truth
+	obsoleted_exclusion_reasons = [x for x in exclusion_dicts if x not in set(df['codes'])]
```

Comment:
FYI: Deletes any old exclusion reasons not found in the [GoogleSheet](https://docs.google.com/spreadsheets/d/18V-_j4b22LzCytxFEM3pBEl6-_ODJY06gpUwH0F6JVY/edit#gid=1644570180).